libpromises/fncall.c

```diff
@@ -244,10 +248,9 @@ static FnCallResult CallFunction(EvalContext *ctx, const Policy *policy, const F
         rp = rp->next;
     }
 
-    char output[CF_BUFSIZE];
     if (argnum != RlistLen(expargs) && !(fncall_type->options & FNCALL_OPTION_VARARG))
     {
-        snprintf(output, CF_BUFSIZE, "Argument template mismatch handling function %s(", fp->name);
+        fprintf(stderr, "Argument template mismatch handling function %s(", fp->name);
```

Comment:
I followed the intent of the previous code. I suggest merging this as is and then fixing it as the developers see fit. Clearly it wasn't a big deal before, so the fix can be delayed until this PR is merged.